src/views/1_ApplicationComponent.tsx

```diff
@@ -133,37 +134,54 @@ export class ApplicationComponent extends React.Component<{}, {}> {
             );
         }
 
-        let sessions = _.flatten(this.activeTab.sessions.map((session, index) => {
-                const isActive = session === this.activeTab.activeSession();
-
-                const sessionComponent = (
-                    <SessionComponent session={session}
-                                      key={session.id}
-                                      style={css.session(isActive, index)}
-                                      isActive={isActive}
-                                      updateStatusBar={isActive ? () => this.forceUpdate() : undefined}
-                                      activate={() => {
-                                   this.activeTab.activateSession(session);
-                                   this.setState({ sessions: this.activeTab.sessions });
-                              }}>
-                    </SessionComponent>
-                );
-
-                if (isActive) {
-                    return [sessionComponent];
-                } else {
-                    return [
-                        sessionComponent,
-                        <div className="shutter" style={css.sessionShutter(index)}></div>,
-                    ];
-                }
+        let renderSessionContainer: any;
```

Comment:
This mapping will still be necessary in the future to know what session is located, say, above current. But we can delay it, if you don't feel like doing that.